### PR TITLE
Add dependencies

### PR DESCRIPTION
**Pull request by Catatonica**
Add RecyclerView widget, add dimens.xml, remove ConstraintLayout dependency

Add recycle_view_item.xml

Add RecyclerAdapter class with ItemClickListener interface

Add RecyclerView's creation in MainActivity

Add some annotations

### CODE REVIEW

File: app/src/main/res/layout/activity_main.xml [hunk 1]
```diff
@@ -1,18 +1,15 @@
 <?xml version="1.0" encoding="utf-8"?>
-<android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:tools="http://schemas.android.com/tools"
```
Comment (PavelDolbik) [inline app/src/main/res/layout/activity_main.xml]: Why you need LinearLayout here ? 
What is happen if you delete it ?

File: app/src/main/res/layout/recycler_view_item.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        android:padding="@dimen/default_padding">
+
+        <TextView
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="@string/item"
+            android:textSize="@dimen/default_text_size" />
+
+        <TextView
+            android:id="@+id/tvItemId"
```
Comment (PavelDolbik) [inline app/src/main/res/layout/recycler_view_item.xml:26]: Why you need it?

Comment (pasha656) [reply app/src/main/res/layout/recycler_view_item.xml]: Why you using LinearLayout here ?

File: app/src/main/res/layout/recycler_view_item.xml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    android:id="@+id/cvContact"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    app:cardCornerRadius="15dp"
+    app:cardElevation="3dp"
+    app:cardMaxElevation="5dp"
+    app:cardUseCompatPadding="true">
+
+    <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+        android:layout_width="match_parent"
```
Comment (PavelDolbik) [inline app/src/main/res/layout/recycler_view_item.xml:13]: Why you using LinearLayout here ?

File: app/src/main/java/com/epam/havina/task1_recycler/MainActivity.java [hunk 4]
```diff
@@ -1,13 +1,38 @@
...
+ * @author Lizaveta Havina
+ * @version 11.10.2018
+ */
+public class MainActivity extends AppCompatActivity implements RecyclerAdapter.ItemClickListener {
+
+    private static final int[] itemsArray = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12};
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
```
Comment (PavelDolbik) [inline app/src/main/java/com/epam/havina/task1_recycler/MainActivity.java]: Please, do not forget about final (final Bundle savedInstanceState)

File: app/src/main/java/com/epam/havina/task1_recycler/MainActivity.java [hunk 5]
```diff
@@ -1,13 +1,38 @@
...
         setContentView(R.layout.activity_main);
+
+        setRecyclerView();
+    }
+
+    @Override
+    public void onItemClicked(@NonNull String itemId) {
```
Comment (PavelDolbik) [inline app/src/main/java/com/epam/havina/task1_recycler/MainActivity.java]: Please, add final

File: app/src/main/java/com/epam/havina/task1_recycler/MainActivity.java [hunk 6]
```diff
@@ -1,13 +1,38 @@
...
+        Toast.makeText(this, itemId + getString(R.string.toast_message), Toast.LENGTH_SHORT).show();
+    }
```
Comment (PavelDolbik) [inline app/src/main/java/com/epam/havina/task1_recycler/MainActivity.java]: Is it good idea using concat string ?

File: app/src/main/java/com/epam/havina/task1_recycler/MainActivity.java [hunk 7]
```diff
@@ -1,13 +1,38 @@
...
+
+    private void setRecyclerView() {
+        RecyclerView recyclerView = findViewById(R.id.recyclerView);
```
Comment (PavelDolbik) [inline app/src/main/java/com/epam/havina/task1_recycler/MainActivity.java]: final

File: app/src/main/java/com/epam/havina/task1_recycler/RecyclerAdapter.java [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+        holder.tvItemId.setText(String.valueOf(itemId));
+    }
+
+    @Override
+    public int getItemCount() {
+        return itemsList.length;
+    }
+
+    class ItemHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
+        private TextView tvItemId;
```
Comment (PavelDolbik) [inline app/src/main/java/com/epam/havina/task1_recycler/RecyclerAdapter.java]: final

File: app/src/main/java/com/epam/havina/task1_recycler/RecyclerAdapter.java [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    public ItemHolder onCreateViewHolder(@NonNull final ViewGroup parent, final int viewType) {
+        return new ItemHolder(LayoutInflater
+                .from(parent.getContext())
+                .inflate(R.layout.recycler_view_item, parent, false));
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull final ItemHolder holder, final int position) {
+        int itemId = itemsList[position];
+        holder.tvItemId.setText(String.valueOf(itemId));
```
Comment (PavelDolbik) [inline app/src/main/java/com/epam/havina/task1_recycler/RecyclerAdapter.java]: Please, use -  holder.getAdapterPosition()

File: app/src/main/java/com/epam/havina/task1_recycler/RecyclerAdapter.java [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+    RecyclerAdapter(@NonNull final ItemClickListener listener,
+                    @NonNull final int[] itemsList) {
+        this.listener = listener;
+        this.itemsList = itemsList;
+    }
+
+    @NonNull
+    @Override
+    public ItemHolder onCreateViewHolder(@NonNull final ViewGroup parent, final int viewType) {
+        return new ItemHolder(LayoutInflater
```
Comment (PavelDolbik) [inline app/src/main/java/com/epam/havina/task1_recycler/RecyclerAdapter.java]: I would like you to suggest this solution: 

`final View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.recycler_view_item, parent, false);

        final RecyclerView.ViewHolder holder = new ItemHolder(view);
        view.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                final int adapterPosition = holder.getAdapterPosition();
                if (adapterPosition != RecyclerView.NO_POSITION) {
                    listener.onItemClicked(itemsList[adapterPosition]);
                }
            }
        });

        return (ItemHolder) holder;`

File: app/src/main/res/layout/activity_main.xml (hunk 1 above)
Comment (Catatonica) [reply app/src/main/res/layout/activity_main.xml]: Well, I don't really need it in this case, only if I'd like to extend activity_main and add more views and containers in future. I can safely delete it.

File: app/src/main/res/layout/recycler_view_item.xml (hunk 2 above)
Comment (Catatonica) [reply app/src/main/res/layout/recycler_view_item.xml:26]: To show item's id.

File: app/src/main/res/layout/recycler_view_item.xml (hunk 3 above)
Comment (Catatonica) [reply app/src/main/res/layout/recycler_view_item.xml:13]: To order TextViews and make them set one after another horizontally, to set common padding and gravity.

File: app/src/main/java/com/epam/havina/task1_recycler/MainActivity.java (hunk 6 above)
Comment (Catatonica) [reply app/src/main/java/com/epam/havina/task1_recycler/MainActivity.java]: No, Android Studio will offer you to replace it with "+" expression in order to increase code clarity and performance

**Pull request merged**